src/Runner.Worker/ActionDepthOutOfRange.cs

```diff
@@ -0,0 +1,22 @@
+using System;
```

Comment:
nit: in general we don't have specific exceptions and this is going to increase the files to grok in the Action.Worker directory. Consider just asserting the string in the unit test

Looks like ActionNotFoundException is the only one, and it's the outlier not the predominant pattern

Reply:
It looks like ActionNotFound is explicitly caught so it's not retried

Reply:
AssertThrows returns the exception which can then be used to check the message